atomisticparsers/utils/parsers.py

```diff
@@ -179,52 +179,53 @@ def parse_md_workflow(self, data: Dict[str, Any]) -> None:
         self.archive.workflow2 = sec_workflow
 
     def parse_interactions(self, interactions: List[Dict], sec_model: MSection) -> None:
-        interaction_dict = {}
-        for interaction_key in Interaction.m_def.all_quantities.keys():
-            interaction_dict[interaction_key] = np.array(
-                [interaction.get(interaction_key) for interaction in interactions],
-                dtype=object,
-            )
-        interaction_dict = {key: val for key, val in interaction_dict.items()}
-        interaction_types = (
-            np.unique(interaction_dict["type"])
-            if interaction_dict.get("type") is not None
-            else []
-        )
-        for interaction_type in interaction_types:
+        if not interactions:
+            return
+
+        def write_interaction_values(values):
             sec_interaction = Interaction()
             sec_model.contributions.append(sec_interaction)
-            interaction_indices = np.where(
-                interaction_dict["type"] == interaction_type
-            )[0]
-            sec_interaction.type = interaction_type
-            sec_interaction.n_interactions = len(interaction_indices)
-            sec_interaction.n_atoms
-            for key, val in interaction_dict.items():
+            sec_interaction.type = current_type
+            sec_interaction.n_atoms = max(
+                [len(v) for v in values.get("atom_indices", [[0]])]
+            )
+            for key, val in values.items():
+                # TODO tempory fix: atom_labels, atom_indices not homogeneous
+                # fill in missing atom label with 'X', atom index with -1
```

Comment:
I don't think it is actually what the function below is doing, and I don't understand what you mean by fill in missing values? I guess you are thinking a case where atom_indices and atom_labels are of a different length than the actual list of atoms? This would be a very strange occurrence from my perspective and should result in an error, unless we have a concrete use case for this.

Reply:
From the denition atom_labels and atom_indices have shape ['n_interactions', 'n_atoms'] but n_atoms is not the same for all interactions. Is this right? Hence the fix that I have here that fulfills the definition is to set n_atoms to the max value  l189 and for interactions with n-atoms less than max value, I fill in 'X' or -1 respectively. As I have mentioned above, this is not an ideal fix and it is better to redefine Interaction section so we do not group all interactions.